plugins/flow-visibility/clickhouse-monitor/main.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// The storage percentage at which the monitor starts to delete old records. By default, if the storage usage is larger than 50%, it starts to delete the old records.
+	threshold = 0.5
+	// The percentage of records in Clickhouse will be deleted when the storage is above threshold.
+	deletePercentage = 0.5
+	// The monitor stops for 3 intervals after a deletion to wait for the Clickhouse MergeTree Engine to release memory.
+	skipRoundsNum = 3
+	// The name of the table to store the flow records
+	tableName = "default.flows"
+)
+
+func main() {
+	// The monitor stops working for several rounds after a deletion
+	// as the release of the memory space for Clickhouse MergeTree engine requires time
+	if !skipRound() {
+		connect, err := connectLoop()
+		if err != nil {
+			klog.Error(err)
+			return
+		}
+		deleted := monitorMemory(connect)
+		if deleted {
+			klog.Infof("Number of rounds to be skipped: %d", skipRoundsNum)
+		} else {
+			klog.Info("Number of rounds to be skipped: 0")
+		}
+	}
+}
+
+// Checks the k8s log for the number of rounds to skip.
+// Returns true when the monitor needs to skip more rounds and logs the number of rounds to skip for next time,
+// Otherwise returns false.
+func skipRound() bool {
+	logString, err := getPodLogs()
+	if err != nil {
+		klog.Errorf("Error in finding last monitor job: %v", err)
+		return false
+	}
+	// reads the number of rounds requires to be skipped
+	logs := strings.Split(logString, "Number of rounds to be skipped: ")
+	if len(logs) < 2 {
+		klog.Errorf("Error in finding last monitor job: %v", err)
+		return false
+	}
+	lines := strings.Split(logs[1], "\n")
```

Comment:
Could you explain a bit here about why split logs[1] instead of logs[-1]? I think we need to find the last skip round number in logString.

Reply:
As Cronjob launches the monitor in a new pod every minute, each log records the current round, which only contains one line indicating the skip round number. Thus logs[1] is the same as logs[-1] in this case. Maybe I can add a check for the logs length to make it clearer.

Reply:
I feel it might be helpful if you can add a comment here giving a sample `logString`. And explaining it will only contain a single line.

Reply:
Added